File: msg-entity/src/main/java/dev/number6/entity/dagger/ChannelMessagesHandlerModule.java
package dev.number6.entity.dagger;

import dev.number6.comprehend.results.PresentableEntityResults;
import dev.number6.message.ChannelMessagesComprehensionHandler;
import dev.number6.message.ChannelMessagesHandler;
import dev.number6.message.ChannelMessagesToComprehensionResultsFunction;
import dev.number6.message.ComprehensionResultsConsumer;
import dagger.Module;
import dagger.Provides;

@Module
public class ChannelMessagesHandlerModule {

    @Provides
    public ChannelMessagesHandler providesMessageComprehension(ChannelMessagesToComprehensionResultsFunction<PresentableEntityResults> channelMessagesToComprehensionResultsFunction,
                                                               ComprehensionResultsConsumer<PresentableEntityResults> comprehensionResultsConsumer) {
        return new ChannelMessagesComprehensionHandler<>(channelMessagesToComprehensionResultsFunction, comprehensionResultsConsumer);
    }
}

File: msg-entity/src/main/kotlin/dev/number6/entity/dagger/ChannelMessagesHandlerModule.kt
package dev.number6.entity.dagger

import dagger.Module
import dagger.Provides
import dev.number6.comprehend.results.PresentableEntityResults
import dev.number6.message.ChannelMessagesComprehensionHandler
import dev.number6.message.ChannelMessagesHandler
import dev.number6.message.ChannelMessagesToComprehensionResultsFunction
import dev.number6.message.ComprehensionResultsConsumer

@Module
class ChannelMessagesHandlerModule {
    @Provides
    fun providesMessageComprehension(channelMessagesToComprehensionResultsFunction: ChannelMessagesToComprehensionResultsFunction<PresentableEntityResults>,
                                     comprehensionResultsConsumer: ComprehensionResultsConsumer<PresentableEntityResults>): ChannelMessagesHandler {
        return ChannelMessagesComprehensionHandler<PresentableEntityResults>(channelMessagesToComprehensionResultsFunction, comprehensionResultsConsumer)
    }
}